sbt-metadoc/src/sbt-test/metadoc/hello-world/build.sbt

```diff
@@ -47,9 +47,4 @@ TaskKey[Unit]("check") := {
       s"Symbol does not exist: $symbol"
     )
 
-  val actualSymbols = (dir / "symbol").listFiles().length
-  assert(
-    actualSymbols == expectedSymbols.length,
-    s"Found $actualSymbols symbols, expected ${expectedSymbols.length}"
-  )
```

Comment:
Some like the following will make it possible to guess unexpected symbols:

```scala
  val actualSymbols = (dir / "symbol").listFiles().map(_.getName).toSet
  val symbolsDiff = actualSymbols -- expectedSymbols.map(encodeSymbolName).toSet

  assert(
    symbolsDiff.isEmpty,
    symbolsDiff.map { symbol =>
      val name = IO.readLines(dir / "symbol" / symbol).find(_.nonEmpty).getOrElse("???")
      s"$symbol\n - resolves to: $name"
    }.mkString("Found unexpected symbols:\n", "\n", "")
  )
```

This will make the test fail with:

```
[info] java.lang.AssertionError: assertion failed: Found unexpected symbols:
[info] e3ae759f334dc7af9d0b1ded739317abc5bbe3ed4737705f98aada92690276a05a445723450cfa05e51cc8e0e332d0b5c230516b068f3fe4094830b7e300835e
[info]  - resolves to: #_root_.hello.HelloTest#`<init>`()V.$
```

I assume that the new symbol `_root_.hello.HelloTest#`<init>`()V.$` is the HelloTest object initializer?

Reply:
I would prefer to keep the scripted test as small as possible and test behavior of the cli in `tests/`. Scripted is incredibly slow to run and I generally dislike writing advanced logic inside build.sbt files of scripted project (one reason: bad IntelliJ support).

> I assume that the new symbol _root_.hello.HelloTest#()V.$ is the HelloTest object initializer?

Correct, the `()V.` part means "unit returning method".

Reply:
Fair enough. How about we copy the semantic DB from this scripted test into the tests code and do the assertions there?

Reply:
Agreed, the cli definitely could benefit from more tests. I opened https://github.com/scalameta/metadoc/issues/60 to track it.